geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SimpleFeatureTypeLoader.scala

```diff
@@ -28,91 +23,65 @@ import scala.collection.JavaConverters._
-      confs.root.keySet.flatMap { name =>
-        val sftConf = confs.getConfig(name)
+  /**
+    * Load types from arbitrary urls
+    */
+  class URLSftProvider extends SimpleFeatureTypeProvider with ConfigSftParsing {
+    override def loadTypes(): java.util.List[SimpleFeatureType] = {
+      val urls = configURLs.toList
+      logger.debug(s"Loading config from urls: ${urls.mkString(", ")}")
+      val configs = urls.flatMap { url =>
+        logger.debug(s"Attempting to parse config from url $url")
         try {
-          val sft = SimpleFeatureTypes.createType(sftConf, Some(name))
-          Some(sft)
+          Some(ConfigFactory.parseURL(url))
         } catch {
-          case e: Exception =>
-            logger.error("Error loading simple feature type from config " +
-              s"${sftConf.root().render(ConfigRenderOptions.concise())}", e)
+          case e: Throwable =>
+            logger.warn(s"Unable to load SFT config from url $url")
+            logger.trace(s"Unable to load SFT config from url $url", e)
             None
         }
-      }.toList.asJava
+      }
+      configs.reduceLeftOption(_.withFallback(_)) match {
+        case Some(c) => parseConf(c).asJava
+        case None => Collections.emptyList[SimpleFeatureType]()
+      }
     }
-  }
-}
-
-object ConfigSftParsing {
-  val ConfigPathProperty = SystemProperty("org.locationtech.geomesa.sft.config.path", "geomesa.sfts")
 
-  // keep as function so its mutable
-  def path: String = ConfigPathProperty.get
-}
-
-class ClassPathSftProvider extends SimpleFeatureTypeProvider with ConfigSftParsing {
-  override def loadTypes(): JList[SimpleFeatureType] = {
-    val sfts = parseConf(ConfigFactory.load())
-    logger.debug(s"Loading SFTs from classpath ${sfts.map(_.getTypeName).mkString(", ")}")
-    sfts
-  }
-}
-
-class URLSftProvider extends SimpleFeatureTypeProvider with ConfigSftParsing {
-  import URLSftProvider._
-  override def loadTypes(): JList[SimpleFeatureType] = {
-    val urls = configURLs.toList
-    logger.debug(s"Loading config from urls: ${urls.mkString(", ")}")
-    urls.flatMap { url =>
-      logger.debug(s"Attempting to parse config from url $url")
-      try {
-        Some(ConfigFactory.parseURL(url))
-      } catch {
-        case e: Throwable =>
-          logger.warn(s"Unable to load SFT config from url $url")
-          logger.trace(s"Unable to load SFT config from url $url", e)
-          None
-      }
-    }.reduceLeftOption(_.withFallback(_))
-    .map(parseConf)
-    .getOrElse(List.empty[SimpleFeatureType])
-  }
-  
-  // Will also pick things up from the SystemProperties
-  def configURLs: Seq[URL] = {
-    val config = ConfigFactory.load()
-    if (config.hasPath(SftConfigURLs)) {
-      config.getAnyRef(SftConfigURLs) match {
-        case s: String          => s.split(',').map(s => s.trim).toList.map(new URL(_))
-        case lst: JList[String] => lst.map(new URL(_))
+    // Will also pick things up from the SystemProperties
```

Comment:
Are all the ways of applying substitutions documented?  Would it be easy to add that to the docs?

Reply:
do you mean HOCON substitutions? we don't explicitly document it, but it's in the HOCON docs themselves.

Reply:
which we link to, in some places at least